### PR TITLE
resolving documentation warnings

### LINKED ISSUE #717: fix docs formatting
## 📚 Documentation

At this moment the docs raise about 140 warnings which may cause some issues in documentation.
To build the docs locally follow these steps:
```bash
cd docs
pip install -r requirements.txt
make html
```

Thanks!

### PR DESCRIPTION
**Pull request by hanbyul-kim**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [X] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes #717  (issue).
remove warnings (number: 142 -> 9)

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@hanbyul-kim could you pls check the lint issues:
```
./pytorch_lightning/core/lightning.py:355: [E501] line too long (172 > 120 characters)
./pytorch_lightning/core/lightning.py:357: [E501] line too long (207 > 120 characters)
./pytorch_lightning/trainer/distrib_parts.py:278: [E501] line too long (152 > 120 characters)
./pl_examples/domain_templates/gan.py:25: [E302] expected 2 blank lines, found 1
./pl_examples/domain_templates/gan.py:196: [F821] undefined name 'pl'
./pl_examples/basic_examples/lightning_module_template.py:21: [E302] expected 2 blank lines, found 1
```

**Comment by hanbyul-kim** [comment]
> @hanbyul-kim could you pls check the lint issues:
> 
> ```
> ./pytorch_lightning/core/lightning.py:355: [E501] line too long (172 > 120 characters)
> ./pytorch_lightning/core/lightning.py:357: [E501] line too long (207 > 120 characters)
> ./pytorch_lightning/trainer/distrib_parts.py:278: [E501] line too long (152 > 120 characters)
> ./pl_examples/domain_templates/gan.py:25: [E302] expected 2 blank lines, found 1
> ./pl_examples/domain_templates/gan.py:196: [F821] undefined name 'pl'
> ./pl_examples/basic_examples/lightning_module_template.py:21: [E302] expected 2 blank lines, found 1
> ```

Sure. I've been busy this weekend. I'll come back and do it tonight.

**Comment by hanbyul-kim** [comment]
@Borda I can't build docs anymore locally. Do you have any idea about this?

Partial import of `torchlightning` during the build process.
WARNING: while setting up extension sphinxcontrib.mockautodoc: directive 'automodule' is already registered, it will be overridden

Sphinx error:
Builder name install not registered or available through entry point
make: *** [install] Error 2

**Comment by Borda** [comment]
@hanbyul-kim Could you please rebase on the master, I will check the build...

**Comment by Borda** [comment]
Well it works for me and also the CI pass docs build
```
done
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 140 warnings.

The HTML pages are in build/html.
```

**Comment by hanbyul-kim** [comment]
@Borda Hi good morning. I think I managed to pass all the tests. Please review my code and approve it.  😆

**Comment by hanbyul-kim** [comment]
@neggert Hi, I made a change to resolve warning issues in docs with help of @Borda. Could you please review my changes?

File: docs/source/index.rst [hunk 1]
```diff
@@ -68,6 +68,7 @@ PyTorch-Lightning Documentation
    CODE_OF_CONDUCT.md
    CONTRIBUTING.md
    BECOMING_A_CORE_CONTRIBUTOR.md
+   PULL_REQUEST_TEMPLATE.md
```
Comment (Borda) [inline docs/source/index.rst]:
```suggestion
```

**Comment by Borda** [comment]
@williamFalcon this also fixes missing examples in actual docs
see: https://pytorch-lightning.readthedocs.io/en/latest/pl_examples.domain_templates.gan.html

**Comment by Borda** [comment]
moreover, this shall prevent future issues #844

**Comment by williamFalcon** [comment]
```
================================================================================================= FAILURES =================================================================================================
_______________________________________________________________________________________________ FLAKE8-check _______________________________________________________________________________________________
/home/waf251/Developer/pytorch-lightning/pytorch_lightning/trainer/trainer.py:30: [F811] redefinition of unused 'LightningModule' from line 21
```

**Comment by williamFalcon** [comment]
Failing GPU tests...
Mind fixing and/or rebasing master?

```
_____________________________________________________________________________________________ test_amp_gpu_ddp _____________________________________________________________________________________________

tmpdir = local('/tmp/pytest-of-waf251/pytest-6/test_amp_gpu_ddp0')

    def test_amp_gpu_ddp(tmpdir):
        """Make sure DDP + AMP work."""
        if not tutils.can_run_gpu_test():
            return

        tutils.reset_seed()
        tutils.set_random_master_port()

        hparams = tutils.get_hparams()
        model = LightningTestModel(hparams)

        trainer_options = dict(
            default_save_path=tmpdir,
            show_progress_bar=True,
            max_epochs=1,
            gpus=2,
            distributed_backend='ddp',
            precision=16
        )

>       tutils.run_model_test(trainer_options, model)

tests/test_amp.py:81:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/models/utils.py:63: in run_model_test
    result = trainer.fit(model)
pytorch_lightning/trainer/trainer.py:904: in fit
    mp.spawn(self.ddp_train, nprocs=self.num_gpus, args=(model,))
../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/site-packages/torch/multiprocessing/spawn.py:162: in spawn
    process.start()
../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/multiprocessing/context.py:283: in _Popen
    return Popen(process_obj)
../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/multiprocessing/popen_spawn_posix.py:32: in __init__
    super().__init__(process_obj)
../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/multiprocessing/popen_spawn_posix.py:47: in _launch
    reduction.dump(process_obj, fp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = <SpawnProcess name='SpawnProcess-3' parent=29993 initial>, file = <_io.BytesIO object at 0x7f962dc853b0>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'ModelCheckpoint.__init__.<locals>.<lambda>'

../../media/falcon_kcgscratch1/software/miniconda3/envs/pl4/lib/python3.8/multiprocessing/reduction.py:60: AttributeError
```

**Comment by hanbyul-kim** [comment]
@williamFalcon I didn't touch that gpu part, so I'm not sure that shoud be resolved this issue. I guess this issue has something to do with docs. How do you think about it?

**Comment by Borda** [comment]
@hanbyul-kim it maybe was there before... pls check the solution here https://github.com/PyTorchLightning/pytorch-lightning/pull/849#issuecomment-589940698

**Comment by hanbyul-kim** [comment]
> @hanbyul-kim it maybe was there before... pls check the solution here [#849 (comment)](https://github.com/PyTorchLightning/pytorch-lightning/pull/849#issuecomment-589940698)

I've just reviewed the issue, and I don't have same structure as the one because it split callback classes from pt_callbacks.py. So if I change the part, it may generate conflict during merging. I guess it's different issue. Do I have to rebase #849 too?

**Comment by Borda** [comment]
well it depends which PR will be merged first, I would do the same change for now https://github.com/PyTorchLightning/pytorch-lightning/commit/2244f8cbaa112808ec3fc51428f523a9ca977abc

**Comment by Borda** [comment]
@hanbyul-kim may you check also suggestions from Sphinx bot, in "Unchanged files with check annotations" bottom page in tab changed files...

**Comment by hanbyul-kim** [comment]
> @hanbyul-kim may you check also suggestions from Sphinx bot, in "Unchanged files with check annotations" bottom page in tab changed files...

Resolved 87ee3e7

**Comment by hanbyul-kim** [comment]
@Borda @williamFalcon Rebased master again and pushed. Please review it again.

**Comment by hanbyul-kim** [comment]
I'm failing CI test.. maybe related to #859 ?

**Comment by hanbyul-kim** [comment]
@williamFalcon I don't have ddp machine, so I can't run the gpu test. However, this branch has rebased and 2244f8c was applied. So I guess there won't be more problems.

**Comment by Borda** [comment]
just tried to rerun checks manually... LGTM @williamFalcon can you run GPU to get it done?

**Comment by Borda** [comment]
@williamFalcon this one starting to be a bit old... could make it done just after @hanbyul-kim rebase it?

File: tests/test_profiler.py [hunk 2]
```diff
@@ -37,7 +37,9 @@ def test_simple_profiler_durations(simple_profiler, action, expected):
 
 
 def test_simple_profiler_overhead(simple_profiler, n_iter=5):
-    """Ensure that the profiler doesn't introduce too much overhead during training."""
+    """
+    ensure that the profiler doesn't introduce too much overhead during training
+    """
```
Comment (Borda) [inline tests/test_profiler.py]:
```suggestion
    """Ensure that the profiler doesn't introduce too much overhead during training."""
```

File: tests/test_profiler.py [hunk 3]
```diff
@@ -47,25 +49,24 @@ def test_simple_profiler_overhead(simple_profiler, n_iter=5):
 
 
 def test_simple_profiler_describe(simple_profiler):
-    """Ensure the profiler won't fail when reporting the summary."""
+    """
+    ensure the profiler won't fail when reporting the summary
+    """
```
Comment (Borda) [inline tests/test_profiler.py]:
```suggestion
    """Ensure the profiler won't fail when reporting the summary."""
```

File: tests/test_profiler.py [hunk 4]
```diff
@@ -4,26 +4,26 @@
...
     profiler = AdvancedProfiler()
     return profiler
 
 
 @pytest.mark.parametrize("action,expected", [("a", [3, 1]), ("b", [2]), ("c", [1])])
 def test_simple_profiler_durations(simple_profiler, action, expected):
-    """Ensure the reported durations are reasonably accurate."""
+    """
+    ensure the reported durations are reasonably accurate
+    """
```
Comment (Borda) [inline tests/test_profiler.py]:
```suggestion
    """Ensure the reported durations are reasonably accurate."""
```

File: .github/workflows/greetings.yml [hunk 5]
```diff
@@ -10,5 +10,5 @@ jobs:
     - uses: actions/first-interaction@v1
       with:
         repo-token: ${{ secrets.GITHUB_TOKEN }}
-        issue-message: 'Hey, thanks for your contribution! Great first issue!'
-        pr-message: 'Hey, thanks for the input! Please give us a bit of time to review it!'
+        issue-message: 'Hi! thanks for your contribution!, great first issue!'
+        pr-message: 'Hey thanks for the input! please give us a bit of time to review it!'
```
Comment (Borda) [inline .github/workflows/greetings.yml]:
```suggestion
        pr-message: 'Hey thanks for the input! Please give us a bit of time to review it!'
```

**Pull request merged**